### PR TITLE
Fix tests

### PR DESCRIPTION
**Pull request by tomw66**
A combination of branches `CAP-292-time-series-obs-data` and `CAP-321-write-tests-for-cubes-py-and-fix-functionality` (plus a tiny test tidy up)

### CODE REVIEW

**Comment by corinnebosley** [comment]
There's a weird error in one of the tests I wrote, seems like nothing you would have changed (it's in tests/unit/visualise/test_dataset_renderer.py) and I don't understand exactly what the problem is.

One thing I have noticed though is that when I am running the tests and I get to this particular test (and one or two others), it takes ages.  I think It's just the ones using Polygon or MultiPolygon, but I can't understand why they take so long to run and I can't understand why this test has failed.

I guess we'll chat about it later.

File: tests/unit/visualise/test_dataset_renderer.py [hunk 1]
```diff
@@ -215,14 +215,11 @@ def test_diurnal_averaged_data(self, multiday_data):
...
-    #     aircraft_track = dr.TimeSeries(aircraft_data).track(start, end)
-    #     assert len(aircraft_track.dim_coords) == 3
+    def test_track_data(self, aircraft_data):
+        """
+        GIVEN a cube of aircraft data
+        WHEN Timeseries.track() is called for valid time bounds
+        THEN an iris cube is returned
+        """
+        track_data = dr.TimeSeries(aircraft_data).track(PartialDateTime(hour=13), PartialDateTime(hour=14))
+        assert isinstance(track_data, Cube)
```
Comment (corinnebosley) [inline tests/unit/visualise/test_dataset_renderer.py:225]: Nice work getting this test working again (and really the functionality behind it).

I do have some edits to request though.  Although this is indeed cut down to its simplest form I find it a little unclear, so I prefer the method of setting start and end PartialDateTimes separately as it is clearer what the test is then.  Also, I would like to see a test checking the dimensionality of the output cube (as the input cube was a cube anyway, so this could be a null test if we don't also check dims).

**Review by corinnebosley — CHANGES_REQUESTED** [review]
All looks good except for the things in the comments I've posted.

File: src/clean_air/data/data_subset.py [hunk 2]
```diff
@@ -114,16 +112,32 @@ def extract_box(self, box, crs=None):
...
+        # remove all coords except for time DimCoord
+        for coord in cube.aux_coords:
+            cube.remove_coord(coord)
+        if len(cube.dim_coords) != 1:
+            print('Found extra dimensions, attempting to remove...')
+            for coord in cube.dim_coords:
+                if coord.standard_name != 'time':
+                    cube.remove_coord(coord)
+
+        return cube
```
Comment (corinnebosley) [inline src/clean_air/data/data_subset.py:140]: This work is beautiful.  I feel much happier with this than the previous incarnation.  Very nice.

File: tests/unit/visualise/test_dataset_renderer.py (hunk 1 above)
Comment (tomw66) [reply tests/unit/visualise/test_dataset_renderer.py:227]: Good ideas, have adjusted as suggested!

File: src/clean_air/data/data_subset.py (hunk 2 above)
Comment (tomw66) [reply src/clean_air/data/data_subset.py:145]: I've adjusted the code so it now works with just one or the other, with accompanying tests in data subset's unit tests.

**Comment by tomw66** [comment]
The issue with the test failing in tox seems to be (of course) related to conda. I've been adjusting package channels and rebuilding environments, but not sure this is fixible without abandoning anaconda entirely:
https://stackoverflow.com/questions/74226436/hdf5-error-when-opening-nc-files-in-python-with-xarray

Short of that, I've basically Ship of Theseus-ed the test to:
- be in a different location
- have a different name
- have a different description
- use completely different source data
- use remade timeseries data
- load a different path 
- test a different property

So basically an entirely new test - but it still fails, whilst all others are fine! I'm at a loss :(
